src/plugins/explore/public/application/register_tabs.ts

```diff
@@ -44,18 +50,32 @@ export const registerBuiltInTabs = (tabRegistry: TabRegistryService) => {
     order: 15,
     supportedLanguages: [EXPLORE_DEFAULT_LANGUAGE],
 
-    prepareQuery: (query) => {
-      return typeof query.query === 'string' ? query.query : '';
+    prepareQuery: (query, state) => {
+      // Get the selected patterns field from the Redux state
+      let patternsField = state?.tab.patterns.patterns_field;
+
+      const preparedQuery = getQueryWithSource(query);
+      if (!patternsField) {
+        patternsField = findDefaultPatternsField(state, services);
+      }
+
+      return typeof preparedQuery.query === 'string' && preparedQuery.query !== ''
+        ? preparedQuery.query +
+            ` | patterns \`${patternsField}\` method=brain mode=aggregation | sort - pattern_count`
+        : preparedQuery.query;
     },
 
     component: PatternsTab,
 
     // Add lifecycle hooks
-    onActive: () => {
-      // Tab activated
+    onActive: (state?: RootState): void => {
+      findDefaultPatternsField(state, services);
     },
-    onInactive: () => {
+
+    onInactive: (state?: RootState): void => {
       // Tab deactivated
+      if (!state || !services?.store) return;
```

Comment:
if state and services are required, i don't think this return would be needed

Reply:
removed